discretisedfield/mesh.py

```diff
@@ -187,22 +182,42 @@ def __init__(
         attributes={"unit": "m"},
     ):
         # TODO NO MUTABLE DEFAULT
-        if region is not None and p1 is None and p2 is None:
-            self.region = region
+        if region is None and p1 is None and p2 is None:
+            raise ValueError("p1, p2, and region cannot be None simultaneously.")
+        elif region is not None and p1 is None and p2 is None:
+            if not isinstance(region, df.Region):
+                raise TypeError("region must be of class discretisedfield.Region.")
+            self._region = region
         elif region is None and p1 is not None and p2 is not None:
-            self.region = df.Region(p1=p1, p2=p2)
+            self._region = df.Region(p1=p1, p2=p2)
         else:
             msg = "Either region or p1 and p2 can be passed, not both."
             raise ValueError(msg)
 
-        if cell is not None and n is None:
-            self.cell = tuple(cell)
-            n = np.divide(self.region.edges, self.cell).round().astype(int)
-            self.n = dfu.array2tuple(n)
+        if cell is None and n is None:
+            raise ValueError("Both cell and n cannot be None.")
+        elif cell is not None and n is None:
+            if not isinstance(cell, Iterable):
+                raise TypeError("Cell must be an Iterable.")
+            self._cell = tuple(cell)
+            if len(self._cell) != self.region.ndim:
+                raise ValueError("The cell must have same dimensions as the region.")
+            elif not all([isinstance(i, (Number, complex)) for i in self._cell]):
```

Comment:
So far we have not had a use case for a complex valued mesh so I do not think it will be a use case that is common at all. Hence, I'm happy not to allow it if it makes our lives easier. However, where does the difficulty come from in supporting these complex meshes? As we are based on `numpy` aren't most of the operations deal with complex numbers automatically.

Reply:
I guess it could cause problems with plotting. Furthermore, proper consistency checks between pmin/pmax and cell could be complicated (e.g. real-valued pmin/pmax with purely imaginary cells do not work, complex-valued pmin/pmax and cell must all consistently have non-zero (or zero) real/imaginary parts, ...).

Reply:
I'll remove the complex check then.